tests/platform_tests/conftest.py

```diff
@@ -91,69 +92,91 @@ def advanceboot_loganalyzer(duthosts, rand_one_dut_hostname, request):
+
+            FMT = "%b %d %H:%M:%S.%f"
+            for _, timings in service_restart_times.items():
+                timestamps = timings["timestamp"]
+                timings["stop_time"] = (datetime.strptime(timestamps["Stopped"], FMT) -\
+                    datetime.strptime(timestamps["Stopping"], FMT)).total_seconds() \
+                        if "Stopped" in timestamps and "Stopping" in timestamps else None
+
+                timings["start_time"] = (datetime.strptime(timestamps["Started"], FMT) -\
+                    datetime.strptime(timestamps["Starting"], FMT)).total_seconds() \
+                        if "Started" in timestamps and "Starting" in timestamps else None
+
+                timings["reboot_time"] = (datetime.strptime(timestamps["Started"], FMT) -\
+                    datetime.strptime(timestamps["Stopped"], FMT)).total_seconds() \
+                        if "Started" in timestamps and "Stopped" in timestamps else None
+
+            files = glob.glob('/tmp/*-report.json')
+            if files:
+                filepath = files[0]
+                with open(filepath) as json_file:
+                    report = json.load(json_file)
+                    service_restart_times.update(report)
+            analyze_result["Services"] = service_restart_times
+            logging.info(json.dumps(analyze_result["Services"], indent=4))
+
+        elif "sairedis.rec" in key:
+            state_times = dict()
+            state_patterns = {
+                "Start creating switch": re.compile(r'.*\|c\|SAI_OBJECT_TYPE_SWITCH.*'),
+                "Finish creating switch": re.compile(r'.*\|g\|SAI_OBJECT_TYPE_SWITCH.*SAI_SWITCH_ATTR_DEFAULT_VIRTUAL_ROUTER_ID.*'),
+                "Set default route": re.compile(r'.*\|S\|SAI_OBJECT_TYPE_ROUTE_ENTRY.*0\.0\.0\.0/0.*SAI_ROUTE_ENTRY_ATTR_PACKET_ACTION=SAI_PACKET_ACTION_FORWARD.*')
+            }
+            states = ["Start creating switch", "Finish creating switch", "Set default route"]
+
+            for message in messages:
+                for state, pattern in state_patterns.items():
+                    if re.search(pattern, message):
+                        timestamp = message.split("|")[0].strip()
+                        state_times[state] = timestamp
+
+            for state in states:
+                logging.info("{:25s} : {}".format(state, state_times[state] if state in state_times else "N/A"))
+
+            analyze_result["sairedis_state"] = state_times
+
     report_file_name = request.node.name + "_report.json"
     report_file_dir = os.path.realpath((os.path.join(os.path.dirname(__file__),\
         "../logs/platform_tests/")))
     report_file_path = report_file_dir + "/" + report_file_name
     if not os.path.exists(report_file_dir):
         os.makedirs(report_file_dir)
     with open(report_file_path, 'w') as fp:
-        json.dump(result, fp, indent=4)
+        json.dump(analyze_result, fp, indent=4)
```

Comment:
Do you have an example of how this json looks like now with sai_redis timings?

Reply:
Added an example output in the PR description.

Reply:
Thanks, I can visualize it better now. In the example that you posted, the switch creation time is <1s. Is that real?

To be consistent with the other timing data in this report, I think we should improve the `sairedis_state` dict a little bit. This will also help when we upload the data to Kusto, and these values can fit well in a database scheman.
One suggestion is as below. But, feel free to make it better as it suits you.
```json
    { "sairedis_state": {
        "sai_switch": {
            "timestamp": {
                "Stopped": "2021-04-23.18:18:51.146830", 
                "Started": "2021-04-23.18:18:51.145539"
            },
            "reboot_time": <difference comes here, which is switch creation time>
        },
        "default_route": {
            "timestamp": {
                "Started": "2021-04-23.18:19:07.419220"
            }
        }
    }
```

Reply:
Updated to make it consistent. Also updated the PR description with the new one. The switch creation data is collected from virtual switch, which is super fast since there is no asic initialization needed.